src/Datagrid/Pager.php

```diff
@@ -71,11 +68,70 @@ public function getQuery()
         return $this->query;
     }
 
+    public function countResults(): int
+    {
+        // NEXT_MAJOR: just return "$this->resultsCount" directly.
+        $deprecatedCount = $this->getNbResults('sonata_deprecation_mute');
+
+        if ($deprecatedCount === $this->resultsCount) {
+            return $this->resultsCount;
+        }
+
+        @trigger_error(sprintf(
+            'Manipulating the protected member "%s::$nbResults" without calling "setResultsCount" is deprecated since sonata-project/doctrine-orm-admin-bundle 3.27 and will fail 4.0.',
```

Comment:
any better idea for a message here? not so happy with that one

Reply:
What about

```
'Relying on the protected member "%s::$nbResults" and its getter/setter is deprecated since sonata-project/doctrine-orm-admin-bundle 3.27 and will fail 4.0. Use "countResults()" and "setResultsCount()" instead.',

Reply:
Seems better indeed

Reply:
Just a little observation: we usually use "property" instead of "member".